src/state/reducers/auth/AuthReducer.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { AUTH_STATUS, AUTH_ACTIONS_KEY } from '../../commons/AuthConstants';
+import { createReducer } from '@reduxjs/toolkit';
+import { Auth } from '@cosmotech/core';
+
+// Authentication data
+export const authInitialState = {
+  id: '',
+  userName: '',
+  profilePic: '',
+  status: AUTH_STATUS.ANONYMOUS
+};
+
+export const authReducer = createReducer(authInitialState, (builder) => {
+  builder
+    .addCase(AUTH_ACTIONS_KEY.REQUEST_LOG_IN, (state, action) => {
+      state.status = AUTH_STATUS.CONNECTING;
+    })
+    .addCase(AUTH_ACTIONS_KEY.REQUEST_LOG_OUT, (state, action) => {
+      Auth.signOut();
```

Comment:
Should not there, should be done in a redux-saga function.
Reducers are pure function (see [reducer](https://redux.js.org/tutorials/fundamentals/part-3-state-actions-reducers#rules-of-reducers))